javascript_nodejs/20.cortana-skill/README.md

```diff
@@ -0,0 +1,54 @@
+This sample demonstrates how to implement a Cortana Skill that properly handles EndOfConversation
+events.
+
+# To try this sample
+- Clone the repository
+    ```bash
+    git clone https://github.com/microsoft/botbuilder-samples.git
+    ```
+- In a terminal, navigate to javascript_nodejs/20.cortana-skill
+    ```bash
+    cd javascript_nodejs/20.cortana-skill
+    ```
+    - Point to the MyGet feed 
+    ```bash
+    npm config set registry https://botbuilder.myget.org/F/botbuilder-v4-js-daily/npm/
+    ```
+- Install modules and start the bot
+    ```bash
+    npm i & npm start
+    ```
+- To reset registry, you can do
+    ```bash
+    npm config set registry https://registry.npmjs.org/
+    ```
+
+# Testing the bot using Bot Framework Emulator
+[Microsoft Bot Framework Emulator](https://github.com/microsoft/botframework-emulator) is
+a desktop application that allows bot developers to test and debug their bots on localhost
+or running remotely through a tunnel.
+
+- Install the Bot Framework Emulator from [here](https://aka.ms/botframework-emulator)
+
+## Connect to bot using Bot Framework Emulator V4
+- Launch Bot Framework Emulator
+- File -> Open Bot Configuration and navigate to javascript_nodejs/20.cortana-skill
+- Select cortana-skill.bot file
+
+# Cortana Skills
+
+Cortana skills are just standard BotBuilder bots that require a few additional considerations specific to Cortana. 
+
+The first thing to understand about a Cortana skill is that Cortana follows a very rigid turn based model of speaking where the user sends a single message to the bot, then the bot sends a single reply to the user, then the user sends a message back to the bot, the bot then sends a reply, and so on. The important thing to note from the bots perspective is that once you've sent a message to the user you are not allowed to send another message to the user until they've replied. You can work around this to some extent using the `inputHint` property off the outgoing activity but in general your skill needs to conform to this back and forth conversation flow.
+
+Another thing unique to Cortana skills is the use of the `EndOfConversation` activity to indicate that the current skill invocation is finished. This activity can be sent from Cortana to the bot to indicate that the user closed the the Cortana window in the UI, and it can be sent from the bot to Cortana to indicate that the Cortana window should be closed. It's worth noting that Cortana in some cases will re-use the same conversation ID in-between skill invocations which can potentially lead to skills starting off in the wrong state so as a best practice your skill should include logic to clear its conversation state anytime an `EndOfConversation` activity is detected. The sample includes a `CortanaSkill` base class that you can derive your bots main dialog from 
```

Comment:
double `the`

Reply:
"in-between" here is awkward.  Maybe say "will re-use the same conversation ID on multiple invocations"